pkg/types/validation/installconfig.go

```diff
@@ -224,6 +224,14 @@ func validateNetworkingIPVersion(n *types.Networking, p *types.Platform) field.E
 		case p.Azure != nil && experimentalDualStackEnabled:
 			logrus.Warnf("Using experimental Azure dual-stack support")
 		case p.BareMetal != nil:
+			apiVIPIPFamily := corev1.IPv6Protocol
+			if net.ParseIP(p.BareMetal.APIVIP).To4() != nil {
+				apiVIPIPFamily = corev1.IPv4Protocol
+			}
+
+			if apiVIPIPFamily != presence["machineNetwork"].Primary {
```

Comment:
This could be simplified to

    if utilnet.IsIPv6String(p.BareMetal.APIVIP) != utilnet.IsIPv6CIDR(n.MachineNetwork[0].CIDR) {

(I think that's right... not tested)

but it looks like nothing else in the installer is using the utilnet (`"k8s.io/utils/net"`) IP family functions currently, so maybe that should be part of a larger future cleanup

/lgtm